overlord/configstate/config/transaction.go

```diff
@@ -172,6 +172,14 @@ func (t *Transaction) Get(snapName, key string, result interface{}) error {
 	// commit changes onto a copy of pristine configuration, so that get has a complete view of the config.
 	config := t.copyPristine(snapName)
 	applyChanges(config, t.changes[snapName])
+
+	out := purgeNulls(config)
+	if out == nil {
```

Comment:
would this happen if we don't normalize away empty maps?

Reply:
We return `NoOptionError` regardless of normalizing empty maps, the idea for this special case here was to make it easier to transform it into 'snap has no configuration'  error if we wanted to, see also the above explanation to @bboozzoo's question. Removing it for now as with the removal of empty maps normalization such case would need different conditonals.